invitations/service.go

```diff
@@ -140,12 +141,31 @@ func (svc *service) AcceptInvitation(ctx context.Context, token, domainID string
 
 		inv.ConfirmedAt = time.Now()
 		inv.UpdatedAt = time.Now()
-		if err := svc.repo.UpdateConfirmation(ctx, inv); err != nil {
-			return err
-		}
+		return svc.repo.UpdateConfirmation(ctx, inv)
 	}
 
-	return nil
+	return svcerr.ErrAuthorization
+}
+
+func (svc *service) RejectInvitation(ctx context.Context, token, domainID string) error {
+	user, err := svc.identify(ctx, token)
+	if err != nil {
+		return err
+	}
+
+	inv, err := svc.repo.Retrieve(ctx, user.GetUserId(), domainID)
+	if err != nil {
+		return err
+	}
+	fmt.Println(inv.UserID)
+
+	if inv.UserID == user.GetUserId() && inv.ConfirmedAt.IsZero() && inv.RejectedAt.IsZero() {
+		inv.RejectedAt = time.Now()
+		inv.UpdatedAt = time.Now()
+		return svc.repo.UpdateRejection(ctx, inv)
+	}
+
+	return svcerr.ErrAuthorization
```

Comment:
Should we return the same error for accepting a rejected invitation as well as one that has already been confirmed?